tests/rptest/tests/license_upgrade_test.py

```diff
@@ -108,3 +108,41 @@ def test_basic_upgrade(self):
 
         # Install license
         assert self.admin.put_license(license).status_code == 200
+
+
+class UpgradeMigratingLicenseVersion(RedpandaTest):
```

Comment:
can we have a follow here on a Matrix annotation that does a to-from upgrades of the last 12 months upgrades to `latest` so ~4 or 5 upgrades.

Reply:
cc: @graphcareful

Reply:
@emaxerrno : AFAIK, we have upgrades from oldest supported release v21.11.x to intervening jumps to latest major's latest minor for the overall upgrade test.

I think what we might need here is to ensure that each such jump is upgrading from all-features-configured cluster to all-feature-in-next-version-cluster and so on.

And as @dotnwat pointed out having the binary compat testing would have caught this as well.